arelle/plugin/validate/ESEF_2022/Util.py

```diff
@@ -0,0 +1,168 @@
+'''
+Created on June 6, 2018
+Version 2022 created on September 19, 2022
+
+Filer Guidelines: https://www.esma.europa.eu/sites/default/files/library/esma32-60-254_esef_reporting_manual.pdf
+
+@author: Workiva
+(c) Copyright 2022 Workiva, All rights reserved.
+'''
+from __future__ import annotations
+import os, json
+
+from arelle.ModelObject import ModelObject
+from arelle.XmlValidate import VALID
+from .Const import esefTaxonomyNamespaceURIs
+from lxml.etree import XML, XMLSyntaxError
+from arelle.FileSource import openFileStream
+from arelle.UrlUtil import scheme
+from arelle.ModelManager import ModelManager
+from arelle.ModelXbrl import ModelXbrl
+from arelle.ValidateXbrl import ValidateXbrl
+from typing import Any, Union, cast
+from arelle.ModelDocument import ModelDocument
+from arelle.typing import TypeGetText
+from collections import defaultdict
+
+
+_: TypeGetText  # Handle gettext
+
+# check if a modelDocument URI is an extension URI (document URI)
+# also works on a uri passed in as well as modelObject
+def isExtension(val: ValidateXbrl, modelObject: ModelObject | ModelDocument | str | None) -> bool:
+    if modelObject is None:
+        return False
+    if isinstance(modelObject, str):
+        uri = modelObject
+    else:
+        uri = modelObject.modelDocument.uri
+    return (uri.startswith(val.modelXbrl.uriDir) or
+            not any(uri.startswith(standardTaxonomyURI) for standardTaxonomyURI in val.authParam["standardTaxonomyURIs"]))
+
+# check if in core esef taxonomy (based on namespace URI)
+def isInEsefTaxonomy(val: ValidateXbrl, modelObject: ModelObject | None) -> bool:
+    if modelObject is None:
+        return False
+    ns = modelObject.qname.namespaceURI
+    return (any(ns.startswith(esefNsPrefix) for esefNsPrefix in esefTaxonomyNamespaceURIs))
```

Comment:
Required for type checking/CI (this is in the other ESEF plugin).
```suggestion
        return False
    assert modelObject.qname is not None
    ns = modelObject.qname.namespaceURI
    assert ns is not None
    return (any(ns.startswith(esefNsPrefix) for esefNsPrefix in esefTaxonomyNamespaceURIs))
```